File: Semester 3/Metode avansate de programare/Curs/MapCurs12-main/ExempleCurs12/ExempleCurs12/domain/Duck.cs
namespace ExempleCurs12.domain;

public class Duck
{
    private int Id { get; set; }
    private string Name {get; set; }
    private double Speed {get; set; }
    private double Resistance {get; set; }

    public Duck(int id, string name, double speed, double resistance)
    {
        this.Id = id;
        this.Name = name;
        this.Speed = speed;
        this.Resistance = resistance;
    }

    public override string ToString()
    {
        return $"{Id} - {Name} - {Speed} - {Resistance}" ;
    }
}

File: Semester 3/Metode avansate de programare/Seminar/sem_8_9_schelet/social-network-CalinNebunu/src/main/java/domain/Duck.java
package domain;

public abstract class Duck extends User {

    private Double viteza;
    private Double rezistenta;
    private Long cardId;
    private String duckType;

    // Constructor
    public Duck(String username, String email, String password, Double viteza, Double rezistenta, Long cardId, String duckType) {
        // Constructor din User
        super(username, email, password);

        this.viteza = viteza;
        this.rezistenta = rezistenta;
        this.cardId = cardId;
        this.duckType = duckType;
    }

    // --- Getters/Setters ---

    public Double getViteza() {
        return viteza;
    }

    public void setViteza(Double viteza) {
        this.viteza = viteza;
    }

    public Double getRezistenta() {
        return rezistenta;
    }

    public void setRezistenta(Double rezistenta) {
        this.rezistenta = rezistenta;
    }

    public Long getCardId() {
        return cardId;
    }

    public void setCardId(Long cardId) {
        this.cardId = cardId;
    }

    public String getDuckType() {
        return duckType;
    }

    public void setDuckType(String duckType) {
        this.duckType = duckType;
    }

    @Override
    public String toString() {
        return String.format("%s Duck {Speed=%.2f, Stamina=%.2f, Card(ID)=%s, Type='%s'}",
                super.toString(),
                viteza,
                rezistenta,
                cardId != null ? cardId : "none",
                duckType);
    }

}
